randomised_quicksort.cpp

```diff
@@ -0,0 +1,59 @@
+#include<iostream>
+#include <bits/stdc++.h>
+#include<stdlib.h>
+#include<time.h>
+using namespace std;
+int partition(int *a,int start,int end)
+{
+    int pindex=0;
+    int pivot=a[end];
+    for(int i=0;i<=end-1;i++)
+    {
+        if(a[i]<=pivot)
+        {
+            swap(a[i],a[pindex]);
+            pindex=pindex+1;
+        }
+    }
+    swap(a[pindex],a[end]);
+    return pindex;
+}
+int random_partition(int *a,int start,int end)  #it will choose the random element as pivot element
+{
+    int r;
+    srand(time(NULL));
+    r=start+rand()%(end - start);
+    swap(a[end],a[r]);
+    return partition(a,start,end);
+}
+
+void quicksort(int *a,int start,int end)
+{
+    if(start<end)
+    {
+        int pindex=random_partition(a,start,end);
+        quicksort(a,start,pindex-1);
+        quicksort(a,pindex+1,end);
+    }
+}
+
+int main()
+{
+    int i;
+    int a[50000];
```

Comment:
instead of statically creating an array of 50,000 elements, use the size input (`n`) to create an array of required size.

```cpp
cin>>n;
int a[n];
```

statically creating an array of 50,000 elements wastes lot of memory.